Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -2503,9 +2514,15 @@ NonnullRefPtr<IfStatement> Parser::parse_if_statement()
         // Code matching this production is processed as if each matching occurrence of
         // FunctionDeclaration[?Yield, ?Await, ~Default] was the sole StatementListItem
         // of a BlockStatement occupying that position in the source code.
+        VERIFY(match(TokenType::Function));
         ScopePusher scope(*this, ScopePusher::Let, Parser::Scope::Block);
         auto block = create_ast_node<BlockStatement>({ m_state.current_token.filename(), rule_start.position(), position() });
-        block->append(parse_declaration());
+        auto function_declaration = parse_declaration();
+        VERIFY(is<FunctionDeclaration>(*function_declaration));
+        auto& func_decl = static_cast<FunctionDeclaration const&>(*function_declaration);
```

Comment:
as mentioned in previous PRs, please avoid abbreviations as much as possible - there's no reason to not spell out `function_declaration` :)

Reply:
This habit seems to be quite hard to kick for me, thanks for the good eye.